sdk/test-utils/recorder/README.md

```diff
@@ -1,9 +1,450 @@
+want to check that everything works fine, you potentially need to run again
+every test. With the common recorder, you can specify what test to run by following Mocha's
+approach of setting certain tests to `it.only`, and also to skip specific tests
+with `it.skip`.  If you launch the recorder in record mode with some of these
+changes (and given that you activate the recorder on `beforeEach`), only the
+files that relate to the changed tests will be updated.  Skipped tests won't
+update their recordings. This way, you can focus on fixing a specific set of
+test with `.only`, then remove all the `.only` calls and trust that the playback will
+keep confirming that the unaffected tests are fine and green.
+
+### Securing sensitive data
+
+Live tests need to do sensitive operations, like authenticating with your Azure
+credentials.  To protect this information from the recordings, the
+test-utils-recorder provides the following functions:
+
+- `setReplaceableVariables`, which will allow you to hide sensitive content
+  from the environment variables (more on that later).
+- `setReplacements`, which will allow you to hide sensitive content by doing
+  pattern matching in the output files.
+- `skipQueryParams`, since query parameters may contain sensitive information,
+  the array provided to method will signal what query parameters to remove from
+  the recordings.
+
+Let's see some examples.
+
+#### setReplaceableVariables
+
+This method will allow you to specify what environment variables have sensitive information.
+Give to this function a plain object composed of the names of the environment
+variables that can't be stored in the recordings, with the values you want to
+store instead. To solve this issue in our previous example test, you may do the following:
+
+```typescript
+// You'll be most certainly importing more than this method.
+import { setReplaceableVariables } from "@azure/test-utils-recorder";
+
+// Do this inside of the beforeEach, before you start recording.
+setReplaceableVariables({
+  AZURE_CLIENT_ID: "azure_client_id",
+  AZURE_CLIENT_SECRET: "azure_client_secret",
+  AZURE_TENANT_ID: "azure_tenant_id",
+  KEYVAULT_NAME: "keyvault_name"
+});
+```
+
+If you record the tests with that change, you should see the values of those
+environment variables replaced with their safe counterpart.
+
+#### setReplacements
+
+A more generic way to clean up your recordings is by providing you with the
+final string of the recording, to which you'll be able to make arbitrary
+changes. The `setReplacements` method expects to receive an array of functions, all of which will receive
+the recorded string and return a string.
+
+Let's say your project can't include the word `mango`. You will be able to get rid of it with the following code inside your `beforeEach`:
+
+```typescript
+setReplacements([
+  (recording: string): string =>
+    recording.replace(/mango/g, "bananas"),
```

Comment:
Love this example. 

[Unrelated to this PR]
Should we consider updating this `setReplacements` method to just take [/mango/g, "bananas"] instead of the whole function?

Reply:
I don't know! I like the idea of a function, but we should probably make an issue to gather feedback from others.